lib/phase/structural.js

```diff
@@ -3,6 +3,37 @@
 const Joi = require('joi');
 const SCHEMA_CONFIG = require('screwdriver-data-schema').config.base.config;
 
+/**
+ * Check if user teardown is at the end of the yaml
+ * @method checkUserTeardown
+ * @param  {Object}          data Screwdriver yaml
+ * @return {Promise}              Error if there is an error
+ */
+function checkUserTeardown(data) {
+    let steps = [];
+    let firstUserTeardownStep = -1;
+    let error = null;
+
+    Object.keys(data.jobs).forEach((job) => {
+        steps = data.jobs[job].steps;
+
+        if (steps) { // if there are user-defined steps
+            for (let i = 0; i < steps.length; i += 1) {
+                const isUserTeardown = Object.keys(steps[i])[0].startsWith('teardown-');
+
+                if (firstUserTeardownStep > -1 && !isUserTeardown) {
+                    error = new Error('User teardown steps need to be at the end');
+                }
+
+                if (isUserTeardown) {
+                    firstUserTeardownStep = i;
```

Comment:
Based on the logic, this `firstUserTeardownStep` seems to be `lastUserTeardownStep`? It will get updated to `teardown-2` if you have 
```yaml
- teardown-1
- teardown-2
```

Reply:
It doesn't matter if you have
```
teardown1
teardown2
```
The intention here is if there is a already a `teardown` and there is another `non-teardown` afterward then it will be error. Otherwise just keep proceeding. I can change the variable name to something else.

Reply:
I feel it's better we change the name to something else, probably without `first`. Otherwise it's kinda confusing